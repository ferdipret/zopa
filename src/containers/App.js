import React, { Component } from "react"
import { connect } from "react-redux"
import { fetchProfile } from "../actions/profile"
import {
  updateInputValue,
  postTransaction,
  fetchTransactions
} from "../actions/transaction"
import Header from "../components/Header"
import Input from "../components/Input"
import Button from "../components/Button"
import Chart from "../components/Chart"
import History from "../components/History"

export class App extends Component {
  componentDidMount() {
    this.getData(1)
  }

  render() {
    const { transaction, profile } = this.props
    return (
      <div>
        <div className="section">
          <Header>Send money</Header>
          <Input
            className="name-input"
            value={transaction.name}
            label="Name"
            onChange={e => this.handleInputChange(e, "name")}
          />
          <Input
            className="email-input"
            value={transaction.email}
            label="Email address"
            onChange={e => this.handleInputChange(e, "email")}
          />
          <Input
            className="amount-input"
            value={transaction.amount.toString()}
            label="Amount"
            onChange={e => this.handleInputChange(e, "amount")}
          />
          <Button className="send-btn" onClick={this.handleSend}>
            Send
          </Button>
        </div>
        <div className="section">
          <Header>My account</Header>
          <Chart
            sent={
              transaction.transactionHistory.length > 0
                ? transaction.transactionHistory.reduce(
                    (a, b) => a + b.amount,
                    0
                  )
                : 0
            }
            available={profile.balance || 0}
          />
          <div>
            <Header>Transactions</Header>
            <History transactions={transaction.transactionHistory} />
          </div>
        </div>
      </div>
    )
  }

  getData = async id => {
    const res = await this.props.fetchProfile(id)
    this.props.fetchTransactions(res.payload.transactionIds)
  }

  handleInputChange = (event, ref) => {
    const { value } = event.target
    this.props.updateInputValue({ value, ref })
  }

  handleSend = () => {
    const { transaction, postTransaction, profile } = this.props
    const { name, email, amount } = transaction

    postTransaction(profile, { name, email, amount })
  }
}

export function mapStateToProps({ transaction, profile }) {
  return {
    transaction,
    profile
  }
}

export default connect(
  mapStateToProps,
  { updateInputValue, postTransaction, fetchProfile, fetchTransactions }
)(App)
