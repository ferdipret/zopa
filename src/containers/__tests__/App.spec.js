import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

describe('App', () => {
  const wrapper = shallow(<App />)

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when clicking the `Send` button', () => {
    it('should call the handleSend classMethod', () => {
      wrapper.find('.send-btn').simulate('click')
    })
  })
})
