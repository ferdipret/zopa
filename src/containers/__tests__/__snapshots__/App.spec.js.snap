// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.div>
          Send money
</styled.div>,
        <Styled(Input)
          label="Name"
/>,
        <Styled(Input)
          label="Email address"
/>,
        <Styled(Input)
          label="Amount"
/>,
        <styled.button
          className="send-btn"
          onClick={[Function]}
>
          Send
</styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Send money",
        },
        "ref": null,
        "rendered": "Send money",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Name",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Email address",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Amount",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Send",
          "className": "send-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Send",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.div>
            Send money
</styled.div>,
          <Styled(Input)
            label="Name"
/>,
          <Styled(Input)
            label="Email address"
/>,
          <Styled(Input)
            label="Amount"
/>,
          <styled.button
            className="send-btn"
            onClick={[Function]}
>
            Send
</styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Send money",
          },
          "ref": null,
          "rendered": "Send money",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Email address",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Amount",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Send",
            "className": "send-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Send",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
